validator/stateless_block_validator.go

```diff
@@ -307,11 +307,17 @@ func SetMachinePreimageResolver(ctx context.Context, mach *ArbitratorMachine, pr
 			if err != nil {
 				log.Error("Failed to deserialize DAS message", "err", err)
 			} else {
-				dasPreimage, err := das.Retrieve(ctx, seqMsg[40:])
+				keysetPreimage, err := das.KeysetFromHash(ctx, cert.KeysetHash[:])
```

Comment:
My understanding of `KeysetFromHash` is that it doesn't guarantee that `keysetPreimage` actually hashes to `cert.KeysetHash` and we check for this at the other callsite. I think we either need to guarantee that in the API or add a equality check here

Reply:
Added a check, treat a mismatch as a keyset preimage fetch failure.